templates/aws/cluster-template.yaml

```diff
@@ -0,0 +1,351 @@
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: ${CLUSTER_NAME}
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks:
+      - 10.1.0.0/16
+    services:
+      cidrBlocks:
+      - 10.152.183.0/24
+  controlPlaneRef:
+    apiVersion: controlplane.cluster.x-k8s.io/v1beta2
+    kind: CK8sControlPlane
+    name: ${CLUSTER_NAME}-control-plane
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
+    kind: AWSCluster
+    name: ${CLUSTER_NAME}
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
+kind: AWSCluster
+metadata:
+  name: ${CLUSTER_NAME}
+spec:
+  region: ${AWS_REGION}
+  sshKeyName: ${AWS_SSH_KEY_NAME}
+  bastion:
+    enabled: ${AWS_CREATE_BASTION:=true}
+  controlPlaneLoadBalancer:
+    healthCheckProtocol: TCP
+  network:
+    cni:
+      cniIngressRules:
+      - description: microcluster
+        protocol: tcp
+        toPort: 2380
+---
+apiVersion: controlplane.cluster.x-k8s.io/v1beta2
+kind: CK8sControlPlane
+metadata:
+  name: ${CLUSTER_NAME}-control-plane
+  namespace: default
+spec:
+  machineTemplate:
+    infrastructureTemplate:
+      apiVersion: infrastructure.cluster.x-k8s.io/v1beta2
+      kind: AWSMachineTemplate
+      name: ${CLUSTER_NAME}-control-plane
+  spec:
+    files:
+      # note(ben): This is only required as long as k8s does not have a stable release.
+      - path: /capi/scripts/install.sh
+        permissions: "0500"
+        owner: "root:root"
+        content: |
+          #!/bin/bash -xe
+          snap install k8s --classic --edge
+      - path: /capi/scripts/bootstrap.sh
+        permissions: "0500"
+        owner: "root:root"
+        content: |
+          #!/bin/bash -xe
+          if [ ! -f /etc/kubernetes/pki/ca.crt ]; then
+            k8s bootstrap --name '{{ ds.meta_data.local_hostname }}' --address "$(cat /capi/etc/microcluster-address)" --file '/capi/etc/config.yaml'
```

Comment:
this is ok to start with, but let's find a better way to specify the name 

the node name should somehow be exposed through the template, we will need similar things in other cloud providers, and it's not feasible to rely on something like this.

Reply:
See https://github.com/canonical/cluster-api-k8s/pull/44